File: 2021/12/10/goal_parser.js
/**
 * https://leetcode.com/problems/goal-parser-interpretation/
 * @author Andrew Jarombek
 * @since 12/10/2021
 */

const interpret = function(command) {
    let result = '';

    let index = 0;
    while (index < command.length) {
        if (command.charAt(index) === 'G') {
            result += 'G';
            index++;
        } else if (command.charAt(index + 1) === ')') {
            result += 'o';
            index += 2;
        } else {
            result += 'al';
            index += 4;
        }
    }

    return result;
};

File: 2022/10/30/java/src/main/java/com/jarombek/GoalParser.java
/*
 * https://leetcode.com/problems/goal-parser-interpretation/
 * @author Andrew Jarombek
 * @since 10/30/2022
 */

package com.jarombek;

class GoalParser {
    public static String interpret(String command) {
        int i = 0;
        StringBuilder sb = new StringBuilder();

        while (i < command.length()) {
            char c = command.charAt(i);

            if (c == 'G') {
                sb.append(c);
                i++;
            } else {
                if (command.charAt(i + 1) == ')') {
                    sb.append("o");
                    i += 2;
                } else {
                    sb.append("al");
                    i += 4;
                }
            }
        }

        return sb.toString();
    }
}
